### ISSUE TITLE
Pass app version to CheckTx

### ISSUE THREAD

**Issue by rootulp** [celestiaorg/celestia-app#2214]
## Context

While working on **Option A** of https://github.com/celestiaorg/celestia-app/issues/2156 @evan-forbes and I realized that CheckTx doesn't have access to the current app version. We want access to the app version so that CheckTx can fetch the upper bound for max square size (a versioned constant) to reject blob txs that are too large.

## Problem

https://github.com/celestiaorg/celestia-app/blob/70b06242669760008428a2b2d265ebfea36990fb/app/check_tx.go#L12-L15

doesn't pass app version. If we want to modify CheckTx in the future in a way that is compatible with single binary syncs, we need a way to branch control flow in CheckTx based on the current block header's app version.

## Proposal

Consider passing app version. Note this likely involves changes to ABCI / Cosmos SDK so this proposal needs further investigation.

**Comment by evan-forbes** [comment]
You're correct that we have to make sure that we have access to the app version, fortunately we should be able to get the app version from the checktx state, which is initialized each block during `Commit` [here](https://github.com/celestiaorg/cosmos-sdk/blob/262703adbb31590538bdb4bc6bf8a5135f816537/baseapp/abci.go#L349-L353)

The general important thing that is missing is we have to pass the app version to all validate basic methods.

**Comment by rootulp** [comment]
Thanks for sharing! The baseApp's [checkState](https://github.com/celestiaorg/cosmos-sdk/blob/262703adbb31590538bdb4bc6bf8a5135f816537/baseapp/baseapp.go#L437-L440) appears private. Do you know which getter we can use to fetch it? I couldn't find one.

**Comment by evan-forbes** [comment]
ahh sorry, by "get from the checktx state", I mean from [that context](https://github.com/celestiaorg/cosmos-sdk/blob/262703adbb31590538bdb4bc6bf8a5135f816537/baseapp/baseapp.go#L439). Which is passed via antehandlers, the rest of the state machine. this might mean that we have to first finish https://github.com/celestiaorg/celestia-app/issues/1166

**Comment by evan-forbes** [comment]
alternatively we could also just use the [`app.AppVersion`](https://github.com/celestiaorg/cosmos-sdk/blob/262703adbb31590538bdb4bc6bf8a5135f816537/baseapp/baseapp.go#L208-L211) method

**Comment by evan-forbes** [comment]
can we close this? or what could close this?

**Comment by rootulp** [comment]
We're no longer doing **Option A** of https://github.com/celestiaorg/celestia-app/issues/2156 so maybe we close this issue for now and open a new issue when we need to plumb app version into various `ValidateBasic()`s.

**Comment by evan-forbes** [comment]
for posterity and to hopefully save a few future braincycles:

the alternative to plumbing app version to validate basics which would be to keep all validate basic method stateless, and then refactor any important checks that would have been performed there to be antehandlers, along with defining a new type so we can define a new validate basic (:grimacing:)

**Comment by tac0turtle** [comment]
can we close this with the multiplexer design?
